modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep_chunks/aah_sweep_chunk.cc

```diff
@@ -39,277 +37,242 @@ AAH_SweepChunk::Sweep(AngleSet& angle_set)
         {
-          const int i = cell_mapping_->MapFaceNode(cf, fi);
+          const int i = cell_mapping.MapFaceNode(f, fi);
+
           for (int fj = 0; fj < num_face_nodes; ++fj)
           {
-            const int j = cell_mapping_->MapFaceNode(cf, fj);
-            const double* psi = sweep_dependency_interface_.GetUpwindPsi(fj);
-            const double mu_Nij = -mu * M_surf_f[i][j];
-            Amat_[i][j] += mu_Nij;
-
-            if (psi == nullptr)
+            const int j = cell_mapping.MapFaceNode(f, fj);
+
+            const double mu_Nij = -face_mu_values[f] * M_surf[f][i][j];
+            Amat[i][j] += mu_Nij;
+
+            const double* psi;
+            if (is_local_face)
+              psi = fluds.UpwindPsi(spls_index, in_face_counter, fj, 0, as_ss_idx);
+            else if (not is_boundary_face)
+              psi = fluds.NLUpwindPsi(preloc_face_counter, fj, 0, as_ss_idx);
+            else
+              psi = angle_set.PsiBndry(cell_face.neighbor_id_,
+                                       direction_num,
+                                       cell_local_id,
+                                       f,
+                                       fj,
+                                       gs_gi,
+                                       gs_ss_begin,
+                                       IsSurfaceSourceActive());
+
+            if (not psi)
               continue;
 
-            for (int gsg = 0; gsg < gs_ss_size_; ++gsg)
-              b_[gsg][i] += psi[gsg] * mu_Nij;
+            for (int gsg = 0; gsg < gs_ss_size; ++gsg)
+              b[gsg][i] += psi[gsg] * mu_Nij;
           } // for face node j
         }   // for face node i
       }     // for f
 
       // Looping over groups, assembling mass terms
-      for (int gsg = 0; gsg < gs_ss_size_; ++gsg)
+      for (int gsg = 0; gsg < gs_ss_size; ++gsg)
       {
-        g_ = gs_gi_ + gsg;
-        gsg_ = gsg;
-        sigma_tg_ = sigma_t[g_];
-
-        const auto& M = *M_;
-        const auto& m2d_op = groupset_.quadrature_->GetMomentToDiscreteOperator();
+        double sigma_tg = sigma_t[gs_gi + gsg];
 
         // Contribute source moments q = M_n^T * q_moms
-        for (int i = 0; i < cell_num_nodes_; ++i)
+        for (int i = 0; i < cell_num_nodes; ++i)
         {
           double temp_src = 0.0;
           for (int m = 0; m < num_moments_; ++m)
           {
-            const size_t ir = cell_transport_view_->MapDOF(i, m, static_cast<int>(g_));
-            temp_src += m2d_op[m][direction_num_] * q_moments_[ir];
-          } // for m
-          source_[i] = temp_src;
-        } // for i
-
-        // Mass Matrix and Source
-        // Atemp  = Amat + sigma_tgr * M
-        // b     += M * q
-        for (int i = 0; i < cell_num_nodes_; ++i)
+            const size_t ir = cell_transport_view.MapDOF(i, m, static_cast<int>(gs_gi + gsg));
+            temp_src += m2d_op[m][direction_num] * source_moments_[ir];
```

Comment:
yep. in m2d, the col index should be moments and the row index should be dir.

Reply:
@wdhawkins , quick question. you marked this as resolved. does this mean that in an upcoming update (on quadratures), the m2d array will be numbered differently and at that momenet, you wil fix also the comment that has ^T ?

Reply:
Yes, that's correct.  I'll address the issue with m2d after I pull it in to the quadrature class and change how it's built. I'll create an issue as a reminder.